src/memray/reporters/flamegraph.py

```diff
@@ -91,12 +108,23 @@ def from_snapshot(
                 else record.stack_trace()
             )
             num_skipped_frames = 0
-            for index, stack_frame in enumerate(reversed(stack)):
+            is_import_system = False
+            for index, (stack_frame, next_frame) in enumerate(
+                pairwise(reversed(stack))
+            ):
                 if is_cpython_internal(stack_frame):
                     num_skipped_frames += 1
                     continue
+                # Check if the next frame is from the import system. We check
+                # the next frame because the "import ..." code will be the parent
+                # of the first frame to enter the import system and we want to hide
+                # that one as well.
```

Comment:
Whoa, do we? You and I were picturing two different things here, then. I thought we were just talking about hiding the frames inside `importlib`, not also hiding the actual call to `import` made from the Python code. The checkbox is labeled "Hide Import System Frames", and I wouldn't consider an `import numpy` statement in `__main__` to be an "import system frame"...

I thought we were just trying to make `import` less distracting and easier to follow by eliding the machinery, not eliding all of the imports entirely.

I wonder if that should be two different check boxes (hide imports vs hide importlib), or if that's overkill...